packages/serialization/text/src/textSerializationWriter.ts

```diff
@@ -19,53 +19,77 @@ export class TextSerializationWriter implements SerializationWriter {
 		this.writeStringValue(key, b64);
 	}
 	private static noStructuredDataMessage = "text does not support structured data";
-	private readonly writer: string[] = [];
+	private readonly writer: (string | null)[] = [];
```

Comment:
why not simply write the null text representation and keep it as an array of strings?